.github/workflows/publish.yaml

```
@@ -0,0 +1,56 @@
+name: Publish
+on: push
+jobs:
+  version-changed:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+      - name: cli
+        id: cli
+        uses: EndBug/version-check@v2
+        with:
+          diff-search: true
+          file-name: ./cli/package.json
+    outputs:
+      cli: ${{ steps.cli.outputs.changed }}
+  publish-cli:
+    runs-on: ubuntu-latest
+    needs: version-changed
+    if: ${{ needs.version-changed.outputs.cli }}
+    env:
+      # Only set 'latest' tag when pushing to main
+      npm_config_tag: ${{ github.ref == 'refs/heads/main' && 'latest' || 'experimental' }}
```

Maybe we can do something like:

```suggestion
      IS_MAIN: ${{ github.ref == 'refs/heads/main' }}
      TAG: ${{ $IS_MAIN && 'latest' || 'experimental' }}
```

---

Don't you think we should just be able to use `YARN_NPM_CONFIG_TAG`?

---

Seems like this is not supported anymore in modern yarn

---

```suggestion
      TAG: ${{ github.ref == 'refs/heads/main' && 'latest' || 'experimental' }}
```

---

Fixed with https://github.com/iota-wiki/iota-wiki/pull/785/commits/6f47e36368319fd1ca3e459d97e8b9639c11aa92 and https://github.com/iota-wiki/iota-wiki/pull/785/commits/2b05a17d9ce5a17c7907128bb8fd796498e8bac5